R/Execution.R

```diff
@@ -41,10 +48,14 @@
 execute <- function(analysisSpecifications,
                     executionSettings,
                     executionScriptFolder = NULL,
+                    keyringName = NULL,
+                    keyringPassword = NULL,
                     restart = FALSE) {
   errorMessages <- checkmate::makeAssertCollection()
+  keyringList <- keyring::keyring_list()
   checkmate::assertClass(analysisSpecifications, "AnalysisSpecifications", add = errorMessages)
   checkmate::assertClass(executionSettings, "ExecutionSettings", add = errorMessages)
+  checkmate::assertLogical(x = (is.null(keyringName) || keyringName %in% keyringList$keyring), add = errorMessages)
```

Comment:
I'm not sure I understand what this check is doing.  assertLogical() tests whether a variable is of type logical. If we want to check whether keyringName is in a list, why not use [checkChoice()](https://rdrr.io/cran/checkmate/man/checkChoice.html)?

Reply:
I'd like to check if the keyringName is NULL (which is the default when calling keyring for the 'system' keyring) or it is in the list of keyring names. Since NULL cannot appear in the vector of keyring names, I opted to make this a logical. I'm open to revising this if you have suggestions here.

Reply:
Nevermind me - I see that the assert* functions support `null.ok = TRUE` so I'll revise to use assertChoice with that option. Thanks!